server/src/test/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsActionTests.java

```diff
@@ -0,0 +1,146 @@
+
+        settingsFilter = new SettingsModule(Settings.EMPTY, Collections.emptyList(), Collections.emptyList()).getSettingsFilter();
+        threadPool = new TestThreadPool("GetSettingsActionTests");
+        clusterService = createClusterService(threadPool);
+        CapturingTransport capturingTransport = new CapturingTransport();
+        transportService = new TransportService(clusterService.getSettings(), capturingTransport, threadPool,
+            TransportService.NOOP_TRANSPORT_INTERCEPTOR,
+            boundAddress -> clusterService.localNode(), null, Collections.emptySet());
+        transportService.start();
+        transportService.acceptIncomingRequests();
+        getSettingsAction = new GetSettingsActionTests.TestTransportGetSettingsAction();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+        threadPool = null;
+        clusterService.close();
+        super.tearDown();
+    }
+    public void testIncludeDefaults() {
+        GetSettingsRequest noDefaultsRequest = new GetSettingsRequest().indices(indexName);
+        getSettingsAction.execute(null, noDefaultsRequest, ActionListener.wrap(noDefaultsResponse -> {
+            assertNull("index.refresh_interval should be null as it was never set", noDefaultsResponse.getSetting(indexName,
+                "index.refresh_interval"));
+        }, exception -> {
+            throw new AssertionError(exception);
+        }));
+
+        GetSettingsRequest defaultsRequest = new GetSettingsRequest().indices(indexName).includeDefaults(true);
+
+        getSettingsAction.execute(null, defaultsRequest, ActionListener.wrap(defaultsResponse -> {
+            assertNotNull("index.refresh_interval should be set as we are including defaults", defaultsResponse.getSetting(indexName,
+                "index.refresh_interval"));
+        }, exception -> {
+            throw new AssertionError(exception);
+        }));
+
+    }
+    public void testIncludeDefaultsWithFiltering() {
+        GetSettingsRequest defaultsRequest = new GetSettingsRequest().indices(indexName).includeDefaults(true)
+            .names("index.refresh_interval");
+        getSettingsAction.execute(null, defaultsRequest, ActionListener.wrap(defaultsResponse -> {
+            assertNotNull("index.refresh_interval should be set as we are including defaults", defaultsResponse.getSetting(indexName,
+                "index.refresh_interval"));
+            assertNull("index.number_of_shards should be null as this query is filtered",
+                defaultsResponse.getSetting(indexName, "index.number_of_shards"));
+            assertNull("index.warmer.enabled should be null as this query is filtered",
+                defaultsResponse.getSetting(indexName, "index.warmer.enabled"));
+        }, exception -> {
+            throw new AssertionError(exception);
+        }));
+    }
+
+    static class Resolver extends IndexNameExpressionResolver {
```

Comment:
out of curiosity, why not use the default resolver?

Reply:
The regular resolver will not return any indices as existing, and further the the hard-coded UUID is needed given the UUID check that happens at Metadata.java:529

I am certainly open to a better way of doing this if one exists

Reply:
the way you do it is not a bad one I think. You could otherwise create a cluster state that contains the indices that you need and make the default resolver work with that one. Not sure it will save you lines of code though and hassle.